src/task.py

```diff
@@ -68,4 +76,5 @@ def list_matched_files(files, format):
     format = re.compile(format)
     files = [file for file in files]
     indexed_files = [(file, format.findall(file)) for file in files]
+    # TODO: Maybe we should raise an error if there are multiple matches
     return [IndexedFile(index[0], file) for file, index in indexed_files if index != []]
```

Comment:
The comment sounds reasonable. I think it may like below code.

```python
class AmbigousIndex(Exception):
    def __init__(self, filename, match_cnt, format_re):
        super().__init__(f'File "{filename}" has "{match_cnt}" matches with format regex "{format_re}"')

def list_matched_files(files, format):
    # Filter the files
    pattern = re.compile(format)
    indexed_files = []
    for file in files:
        match = re.findall(pattern, file)
        if len(match) > 1:
            raise AmbigousIndex(file, len(match), pattern)
        elif len(match) == 1:
            indexed_files.append(IndexedFile(match[0], file))
    return indexed_files
```

Reply:
I have add some comments for this issue. I think under the current implementation, the ambiguous index will never happen. So we don't need to deal with it.

Reply:
I think this is decided by `index regex` design. Now we have index like `[0-9]+`, so there won't be mistakes. If we are sure that `index regex` formats are clean, I agree we don't need to handle it.

Reply:
I think we don't have the responsibility and ability to distinguish the wrong `index regex` from user. Therefore, we don't need to handle it.